### ISSUE TITLE
False positive with `expect(clippy::collapsible_else_if)` inside the `else` block

### ISSUE THREAD

**Issue by schuetzm** [rust-lang/rust-clippy#13365]
### Summary

`expect(clippy::collapsible_else_if)` falsely reports "this lint expectation is unfulfilled" when used inside the `else` block:

### Lint Name

clippy::collapsible_else_if

### Reproducer

```rust
fn main() {
    // Correctly detected as collapsible:
    if true {
    } else {
        if false {}
    }

    // Allowing the lint silences it both inside the `else` block:
    if true {
    } else {
        #[allow(clippy::collapsible_else_if)]
        if false {}
    }

    // ... and before the entire `if`:
    #[allow(clippy::collapsible_else_if)]
    if true {
    } else {
        if false {}
    }

    // Using `expect` inside the `else` block says "this lint expectation is unfulfilled":
    if true {
    } else {
        #[expect(clippy::collapsible_else_if)]
        if false {}
    }

    // This works correctly, however:
    #[expect(clippy::collapsible_else_if)]
    if true {
    } else {
        if false {}
    }
}
```

### Version

```text
rustc 1.81.0 (eeb90cda1 2024-09-04)
binary: rustc
commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
commit-date: 2024-09-04
host: x86_64-unknown-linux-gnu
release: 1.81.0
LLVM version: 18.1.7
```

**Comment by Jarcho** [comment]
This will require the same workaround as `needless_return`. The lint doesn't fire if the inner if has an attribute on it which the causes the lint expectation to be unfulfilled.
